### DISCUSSION TITLE
Extend Permissions to Tenancy Contacts & Roles

### DISCUSSION THREAD

**Discussion by iunderwood** [netbox-community/netbox#8744] [Ideas]
Now that Contacts and Contact Roles are part of the tenancy module, I'm not sure how difficult it would be to add in permissions for those objects, in addition to the Tenant and Tenant Groups.

Thanks in advance!

**Comment by candlerb** [comment]
I see them already (Netbox v3.1.8)

![image](https://user-images.githubusercontent.com/44789/155609830-7520f286-c6bf-4317-abac-ccb6aeb54dd6.png)

**Reply by iunderwood** [reply]
That's what i get for not updating to the new hotness first.  D'oh!

**Reply by iunderwood** [reply]
I dug a little deeper and figured out what prompted me to ask this.

In an earlier verison of NetBox, I added the full list of explicit permissions to an account, prior to these ones being created and available.  As such, I was looking at THOSE, and not the object type list available.

<img width="537" alt="Screen Shot 2022-02-28 at 11 34 11 AM" src="https://user-images.githubusercontent.com/2049749/156021402-37259016-49e5-450c-9ee1-6cf5914d2820.png">

Good times.

**Reply by candlerb** [reply]
If you've been migrating from older versions of Netbox, there was a point where the migration from old to new permissions caused an explosion of separate permissions objects (I had over 300 created IIRC).

Often you can shrink that down to just a handful: for example you can create a single "view all" permission which applies the "Can View" action to a large subset of models (e.g. all of them bar User and Group and a few system-related ones).  But cleaning this up is a manual process.  When a new version of Netbox comes along with new models, you still need to add them to your "view all" permission by hand.
